File: Circular_Queue/Circular_Queue.java
import java.util.Scanner;
class Circular_Queue
{
    public static int front= 0;
    public static int rear= 0;
    public static int queue[] = new int[5];

    static void insertQueue(int item)
    {
        int n =5;
        Circular_Queue.rear = (Circular_Queue.rear+1) % n;

        if(Circular_Queue.front == Circular_Queue.rear)
        {
            System.out.println("Queue is full");
            if(Circular_Queue.rear == 0)
            {
                Circular_Queue.rear = n-1;
            }
            else{
                Circular_Queue.rear = Circular_Queue.rear -1;

            }
        }
        else{
            Circular_Queue.queue[rear] = item;
        }

    }

    static int deleteQueue()
    {
        int n = 5;
        if(Circular_Queue.front == Circular_Queue.rear)
        {
            System.out.println("Queue is empty");
            return -1;
        }
        else{
            Circular_Queue.front = (Circular_Queue.front+1) % n;
            int item = Circular_Queue.queue[Circular_Queue.front];
            Circular_Queue.queue[Circular_Queue.front] = 0;
            return item;
        }
    }

    static void displayQueue(){
        System.out.println("Queue is:\n ");
        for(int i = 0; i<Circular_Queue.queue.length ; i++)
        {
            System.out.println(queue[i]);
        }
    }


    public static void main(String args[])
    {
        System.out.println("Press 1. Insert in the Queue \nPress 2. Delete from the Queue \nPress 3. Display Queue \n Press 4. Exit");
        int choice = 0;
        Scanner sc = new Scanner(System.in);

        while(choice!= 4)
        {
            System.out.println("Enter your choice: ");
            choice = sc.nextInt();

            switch (choice) {
                case 1:
                    System.out.println("Enter the element you want to insert: ");
                    int ele = sc.nextInt();
                    insertQueue(ele);
                    break;

                case 2:
                    int item = deleteQueue();
                    System.out.println("Deleted Value is: " + item);
                    break;
                case 3:
                    displayQueue();
                    break;

                case 4:
                    System.out.println("Thank you :)");
                    break;

                default:
                    break;
            }
        }
    }
}

File: Circular_Queue/Circular_Queue.c
#include <stdio.h>
#include <stdlib.h>
#define MAXSIZE 5
int front=-1,rear=-1,q[MAXSIZE];
void insert(int a){
   if(front==(rear+1)%MAXSIZE) printf("Error!!\nQueue is full!!!\n");
   else{
        if(front==-1) front=0;
        rear =(rear+1)%MAXSIZE;
        q[rear]=a;
        }
}

void delete(){
    if(front==rear){
        printf("Error!! \n Underflow condition\n Queue is empty\n");
    }
    else{
        q[front]=0;
        front=(front+1)%MAXSIZE;
        printf("element successfully delete\n");
        }
}

void display(){
    int i=0;
    for(i=0;i<MAXSIZE;i++){
        printf(" %d  |",q[i]);
    }
    printf("\n");
}

int main(){
    int choise=0,inse;
     printf(" 1) insert element \n 2) delete element \n 3) display queue  \n 4) exit\n");
    while(choise != 4){
    printf(" enter your choice :");
    scanf("%d",&choise);
    switch(choise)
    {
    case 1:
        printf("enter a number which you want  to add in queue:");
        scanf("%d",&inse);
        insert(inse);
        break;
    case 2:
        printf("you chose delete option\n");
        delete();
        break;
    case 3:
        printf("you chose display option\n");
        display();
        break;
    case 4:
        printf("\ni hope u enjoy this prg\n");
        break;
    }
}
return 0;

}


